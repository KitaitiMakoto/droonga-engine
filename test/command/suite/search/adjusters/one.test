# -*- js -*-
#@require-catalog-version 2
#@disable-logging
{
  "type": "add",
  "dataset": "Droonga",
  "body": {
    "table": "Memos",
    "key": "Droonga is fun.",
    "values": {
      "tags": [
        {
          "value": "droonga",
          "weight": 9
        }
      ]
    }
  }
}
{
  "type": "add",
  "dataset": "Droonga",
  "body": {
    "table": "Memos",
    "key": "Groonga is fast!",
    "values": {
      "tags": [
        {
          "value": "groonga",
          "weight": 9
        }
      ]
    }
  }
}
#@enable-logging
{
  "type": "search",
  "dataset": "Droonga",
  "body": {
    "queries": {
      "memos": {
        "source": "Memos",
        "condition": "true",
        "adjusters": [
          {
            "column": "tags",
            "value": "droonga",
            "factor": 5
          }
        ],
        "sortBy": ["-_score"],
        "output": {
          "elements": [
            "records"
          ],
          "limit": 10,
          "attributes": [
            "_key",
            "_score"
          ]
        }
      }
    }
  }
}
