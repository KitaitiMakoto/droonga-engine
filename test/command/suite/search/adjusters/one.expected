[
  "droonga.message",
  0,
  {
    "inReplyTo": "request-id",
    "statusCode": 200,
    "type": "search.result",
    "body": {
      "memos": {
        "records": [
          [
            "Droonga is fun.",
            51
          ],
          [
            "Groonga is fast!",
            1
          ]
        ]
      }
    }
  }
]
